### PR TITLE
Prepare changelog for new release 2024.1.2 

### PR DESCRIPTION
**Pull request by tmichela**

### CODE REVIEW

**Comment by philsmt** [comment]
I think we used to do this everytime for a semesterly release, so LGTM.

File: docs/changelog.md [hunk 1]
```diff
@@ -5,16 +5,18 @@
...
+<!--     in a personal environment you'll need to install the package from git. -->
 
-    ```bash title="Installation command"
-    pip install git+https://github.com/European-XFEL/EXtra.git
-    ```
+<!--     ```bash title="Installation command" -->
+<!--     pip install git+https://github.com/European-XFEL/EXtra.git -->
+<!--     ``` -->
+
+## [2024.2]
```
Comment (JamesWrigley) [inline docs/changelog.md:19]: This needs to have a link to the tag at the bottom of the file (doesn't matter if the tag doesn't exist yet).

**Review by JamesWrigley — CHANGES_REQUESTED** [review]
In hindsight it's a bit confusing to tag releases at the beginning of a new cycle rather than at the end :thinking: Because then someone wanting to install extra for an old experiment in cycle `i` will need to install version `year.{i + 1}` (possibly wrapping around the year) to get any changes that were introduced in the cycle `i`. Should we change to tagging them at the end instead?

**Comment by tmichela** [comment]
That makes sense indeed. Should we then tag that one `2024.1.2` and switch to `2024.2` at the end of this cycle?

**Comment by JamesWrigley** [comment]
Sure, sounds good :+1:

**Review by JamesWrigley — APPROVED** [review]
LGTM!

**Pull request merged**